test/e2e/probe_whitelist_test.go

```diff
@@ -0,0 +1,171 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	authv1alpha1 "istio.io/api/authentication/v1alpha1"
+	v1alpha1authclient "istio.io/client-go/pkg/apis/authentication/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/logstream"
+	rnames "knative.dev/serving/pkg/reconciler/revision/resources/names"
+	"knative.dev/serving/test"
+	v1a1test "knative.dev/serving/test/v1alpha1"
+)
+
+//This test checks if the activator can probe
+//the service when istio end user auth policy is
+//applied on the service.
+//This test needs istio side car injected and
+//istio policy check enabled. If both are not
+//enabled the test will pass
+func TestProbeWhitelist(t *testing.T) {
+	t.Parallel()
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := SetupSideCarEnabledNamespace(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "helloworld",
+	}
+
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer test.TearDown(clients, names)
+
+	t.Log("Creating a new Service")
+	resources, httpsTransportOption, err := v1a1test.CreateRunLatestServiceReady(t, clients, &names, test.ServingFlags.Https)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+
+	url := resources.Route.Status.URL.URL()
+	var opt interface{}
+	if test.ServingFlags.Https {
+		url.Scheme = "https"
+		if httpsTransportOption == nil {
+			t.Fatalf("Https transport option is nil")
+		}
+		opt = *httpsTransportOption
+	}
+	if _, err := pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		url,
+		v1a1test.RetryingRouteInconsistency(pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(test.HelloWorldText))),
+		"HelloWorldServesText",
+		test.ServingFlags.ResolvableDomain,
+		opt); err != nil {
+		t.Fatalf("The endpoint %s for Route %s didn't serve the expected text %q: %v", url, names.Route, test.HelloWorldText, err)
+	}
+
+	// check if side car is injected
+	_, err = getContainer(clients.KubeClient, resources.Service.Name, "istio-proxy", resources.Service.Namespace)
+	if err != nil {
+		t.Log("side car not enabled, skipping test")
+		return
+	}
+
+	// check if deployment is scaled down
+	deploymentName := rnames.Deployment(resources.Revision)
+	if err := WaitForScaleToZeroinNamespace(t, deploymentName, resources.Service.Namespace, clients); err != nil {
+		t.Fatalf("Unable to observe the Deployment named %s scaling down: %v", deploymentName, err)
+	}
+
+	// create auth policy
+	if err = createPolicy(resources.Service.Namespace, clients, resources.Revision.Name); err != nil {
```

Comment:
Is it not possible to add policy config yaml like https://github.com/knative/serving/tree/master/test/config/mtls instead of creating it dynamically? 
We want to know what kind of Policy settings e2e are testing and it could be a good reference for users.

Reply:
Sure. I will make that change.